numpyro/util.py

```diff
@@ -19,6 +21,23 @@ def set_rng_seed(rng_seed):
     onp.random.seed(rng_seed)
 
 
+def set_platform(platform=None):
+    """
+    :param str platform: either 'cpu', 'gpu', 'tpu'.
+    """
+    if platform is None:
+        platform = os.getenv('JAX_PLATFORM_NAME', 'cpu')
+    jax.config.update('jax_platform_name', platform)
+
+
+def set_host_devices(n):
+    """
+    :param int n: number of CPU devices to use.
```

Comment:
You mentioned that setting this changes the default to platform to 'cpu'. Will be good to add that as a warning in the docstring, and also expose these in the docs.

Reply:
Also what happens when we set this to some number which doesn't match the number of actual CPUs we have?

Reply:
Depending on your answer to 1, it might also be worth having a default option based on `os.cpu_count()`.

Reply:
> Will be good to add that as a warning in the docstring

It is not necessary if we set `cpu` as default platform.

> what happens when we set this to some number which doesn't match the number of actual CPUs we have?

I have no idea about this. It just works. You can set it to 1000 even that you just have 10-core CPUs. Because I don't understand it, I don't want to set those as default.

Reply:
> You can set it to 1000 even that you just have 10-core CPUs.

Well, that's strange for sure.